src/main/resources/data/computercraft/lua/rom/programs/poster.lua

```diff
@@ -98,85 +184,90 @@ local function commitPrint(data)
+  term.scroll(4)
+  nY = 15
 end
 
--- Begin printing
-local status = printer.status()
-if status == "busy" then
-  error("Printer is busy", 0)
+local function printRow(sName, sValue, sOptional)
+  term.setTextColor(nHighlightColor)
+  term.write(sName)
+  term.setTextColor(colors.white)
+  term.write(sValue)
+  if sOptional then
+    term.setTextColor(colors.lightGray)
+    term.write(sOptional)
+  end
 end
 
-commitPrint(printData)
-
-term.setTextColor(colors.yellow)
-print("\nHold Ctrl+T to stop\n")
-showPrintStatus()
+-- Loop to print all selected posters
+local nPrinted = 0
+local nMaxPages = #tPosters*nCount
+local bTerminated = false
 
-while true do
-  local e, p1 = os.pullEventRaw()
-  local n = math.min(math.max((count - remaining) + 1, 1), count)
+for i,data in ipairs(tPosters) do
+  local nCur = 0
+  if bTerminated then break end
+  commitPrint(data)
+  
+  term.setCursorPos(1,nY)
+  printRow("Name: ", (data.label or "???"):sub(1,nW-6))
+  
+  term.setCursorPos(1,nY+1)
+  printRow("Status: ", (" %s%% : %s / %s "):format(0, nPrinted, nMaxPages), (" (x%s)       "):format(nCount))
 
-  if e == "poster_printer_complete" or e == "poster_printer_state" then
-    remaining = p1
-    showPrintStatus()
 
-    if remaining <= 0 then
-      -- Printing complete, print the count we managed to print
-      local _, y = term.getCursorPos()
-      term.setCursorPos(1, y)
-      term.clearLine()
-      term.setTextColor(colors.green)
-      print("Printed " .. n .. " item" .. (n ~= 1 and "s" or ""))
+  local cur,max = printer.getInkLevel()
+  term.setCursorPos(1,nY+2)
+  printRow("Ink: ", (" %s%%   "):format(math.floor(100/max*cur)))
+  
+  -- Status and Ink level
+  while true do
+    local e, p1 = os.pullEventRaw()
+    
+    if e == "poster_printer_complete" or e == "poster_printer_state" then
+      local progress = 100-(tonumber(p1) or 0)
+      
+      if progress >= 100 then
+        nPrinted = nPrinted+1
+        nCur = nCur+1
+      end
 
+      term.setCursorPos(1,nY+1)
+      printRow("Status: ", (" %s%% : %s / %s "):format(progress, nPrinted, nMaxPages), (" (x%s)       "):format(nCount))
+      
+      if nCur >= nCount then
+        break
+      end
+    elseif e == "terminate" then
+      printer.stop()
+      bTerminated = true
       break
     end
-  elseif e == "terminate" then    
-    -- Halt printing and print the count we managed to print
-    printer.stop()
-
-    local _, y = term.getCursorPos()
-    term.setCursorPos(1, y)
-    term.clearLine()
-    term.setTextColor(colors.red)
-    print("Printing terminated. Printed " .. n .. " item" .. 
-      (n ~= 1 and "s" or "") .. " out of " .. count)
-
-    break 
   end
 end
 
-term.setTextColor(startColor)
+-- Print results
+term.setCursorPos(1, nY+3)
+if bTerminated then
+  term.setTextColor(nErrColor)
+  print(("Printed %s posters out of %s."):format(nPrinted+1, nMaxPages))
+else
+  term.setTextColor(nGreenColor)
+  print(("Printed %s posters!"):format(nPrinted))
```

Comment:
Not keen on losing proper grammar. This will output "Printed 1 posters!", and "Printed 1 posters out of 3."

Reply:
Also shouldn't these be using %d instead of %s?